cpp/src/Ice/ConnectionFactory.cpp

```diff
@@ -347,9 +349,17 @@ IceInternal::OutgoingConnectionFactory::OutgoingConnectionFactory(
     : _communicator(communicator),
       _instance(instance),
       _monitor(new FactoryACMMonitor(instance, instance->clientACM())),
+      _idleTimeout(60000ms),
+      _enableIdleCheck(false),
       _destroyed(false),
       _pendingConnectCount(0)
 {
+    const PropertiesPtr& properties = _instance->initializationData().properties;
+    int idleTimeout = properties->getPropertyAsIntWithDefault(
+        "Ice.IdleTimeout",
+        static_cast<int>(chrono::duration_cast<chrono::seconds>(_idleTimeout).count()));
+    _enableIdleCheck = properties->getPropertyAsIntWithDefault("Ice.EnableIdleCheck", _enableIdleCheck ? 1 : 0) > 0;
+    _idleTimeout = chrono::duration_cast<chrono::milliseconds>(chrono::seconds(idleTimeout));
```

Comment:
I suggested this in a previous comment but we can avoid the `duration_cast`. It is not required here because the conversion is not lossy.

```suggestion
    _idleTimeout = chrono::milliseconds(chrono::seconds(idleTimeout));
```